cmd/flexible-informer-gen/generators/informer.go

```diff
@@ -182,7 +188,12 @@ func NewFiltered$.type|public$Informer(client $.clientSetInterface|raw$$if .name
 				if tweakListOptions != nil {
 					tweakListOptions(&options)
 				}
-				return client.$.group$$.version$().$.type|publicPlural$($if .namespaced$namespace$end$).Watch(context.TODO(), options)
+				watchObj, err := client.$.group$$.version$().$.type|publicPlural$($if .namespaced$namespace$end$).Watch(context.TODO(), options)
+				if $.isNotFound$(err) {
+					logs.warnf("The version of Istio CRD is not matched with Aspen Mesh", logs.ErrorField(err))
+					return $.newFakeWatch$(), nil
```

Comment:
I'm not as convinced that explicitly checking for the `IsNotFound` error makes as much sense as it does for list; given that `Watch` doesn't seem to leave `Controlplane` spinning (like `List` does), I think letting watch handle that error makes sense. Also, in the case where the proper version of a CRD isn't installed, I believe it will always be handled by the `List` (since informers use `List` to populate the cache initially and then use `Watch`).

Reply:
`Reflector` waits until `ListAndWatch` function doesn't return an error. We can tell here: https://github.com/kubernetes/client-go/blob/release-1.18/tools/cache/reflector.go#L383, once it finishes listing the CRDs it will go to 'watch'. I can demo this to you and Andrew tomorrow.

Reply:
Oh right! I wasn't thinking that we just swallow the error in the initial `List` and so we'd also need a dummy `Watch` if we wanted to make sure that the Controlplane stayed up at all costs. This silently failing makes me a bit uncomfortable, though, especially since the cause is more aggressive than changing a specific resource, but I'll defer to Andrew's judgement.